pandas/core/internals/array_manager.py

```diff
@@ -128,6 +129,11 @@ def __init__(
         # Note: we are storing the axes in "_axes" in the (row, columns) order
         # which contrasts the order how it is stored in BlockManager
         self._axes = axes
+
+        for i, arr in enumerate(arrays):
+            if is_datetime64tz_dtype(arr.dtype) and arr.ndim == 2:
+                assert arr.shape[0] == 1
```

Comment:
When does this happen? In principle, there should only be 1D arrays be passed to the constructor, and ideally should be handled higher up

Reply:
ill take another look and find out

Reply:
need to reshape on the back-end of apply_with_block.  changed locally, will push in a bit